superbench/executor/executor.py

```diff
@@ -122,21 +123,38 @@ def __exec_benchmark(self, context, log_suffix):
             logger.error('Executor failed in %s.', log_suffix)
         return None
 
+    def __get_benchmark_dir(self, benchmark_name):
+        """Get output directory for benchmark's current rank.
+
+        Args:
+            benchmark_name (str): Benchmark name.
+        """
+        benchmark_output_dir = self._output_path / 'benchmarks' / benchmark_name
+        for rank_env in ['PROC_RANK', 'LOCAL_RANK']:
+            if os.getenv(rank_env):
+                benchmark_output_dir /= 'rank{}'.format(os.getenv(rank_env))
+                break
+        return benchmark_output_dir
+
     def __create_benchmark_dir(self, benchmark_name):
         """Create output directory for benchmark.
 
         Args:
             benchmark_name (str): Benchmark name.
         """
-        benchmark_output_dir = self._output_path / 'benchmarks' / benchmark_name
-        if benchmark_output_dir.is_dir() and any(benchmark_output_dir.iterdir()):
-            logger.warning('Benchmark output directory %s is not empty.', str(benchmark_output_dir))
-            for i in itertools.count(start=1):
-                backup_dir = benchmark_output_dir.with_name('{}.{}'.format(benchmark_name, i))
-                if not backup_dir.is_dir():
-                    benchmark_output_dir.rename(backup_dir)
-                    break
-        benchmark_output_dir.mkdir(mode=0o755, parents=True, exist_ok=True)
+        benchmark_output_root_dir = self._output_path / 'benchmarks' / benchmark_name
+        try:
+            if benchmark_output_root_dir.is_dir() and any(benchmark_output_root_dir.iterdir()):
+                logger.warning('Benchmark output root directory %s is not empty.', str(benchmark_output_root_dir))
+                for i in itertools.count(start=1):
+                    backup_dir = benchmark_output_root_dir.with_name('{}.{}'.format(benchmark_name, i))
+                    if not backup_dir.is_dir():
+                        benchmark_output_root_dir.rename(backup_dir)
+                        break
```

Comment:
Can we make it as a util function?  Looks the input can be root_dir and sub_dir.

Reply:
do you mean `__get_benchmark_dir`? the rotation here only handles root dir.

Reply:
I mean the function to check the existence of one dir and rename it. We have the similiar function in sb v0.1

Reply:
updated